src/Microsoft.AspNetCore.Blazor/Components/ParameterAttribute.cs

```diff
@@ -11,5 +11,26 @@ namespace Microsoft.AspNetCore.Blazor.Components
     [AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = false)]
     public sealed class ParameterAttribute : Attribute
     {
+        /// <summary>
+        /// Gets or sets a value to indicate whether the parameter's value should
+        /// be read from the closest ancestor Provider in the tree.
+        ///
+        /// If false (the default), the parameter's value can be supplied only by
+        /// the immediate parent component.
+        /// </summary>
+        public bool FromTree { get; set; }
+
+        /// <summary>
+        /// When <see cref="FromTree"/> is true, gets or sets the name of the
+        /// provided value. When <see cref="FromTree"/> is false, this value is
+        /// not used.
+        ///
+        /// If a value is specified, the parameter value will be supplied by the
+        /// closest ancestor Provider that supplies a value with this name.
+        ///
+        /// If no value is supplied, the parameter value will be supplied by the
+        /// closest ancestor Provider that supplies a value with a compatible type.
+        /// </summary>
+        public string ProviderName { get; set; }
```

Comment:
Based on our discussion yesterday, looking forward to this becoming its own attribute since it won't have much in common with *true* parameters when we're done.

Reply:
Yeah, I think it should be something like `ProviderParameterAttribute`

Reply:
It's now implemented as `CascadingParameterAttribute`, since we decided to use the term "cascading" for this whole feature area.